src/btree/bt_delete.c

```diff
@@ -293,19 +293,43 @@ __wt_delete_page_rollback(WT_SESSION_IMPL *session, WT_REF *ref)
  *     IDs in the delete info. Since we're called at the end of recovery there's no need to lock the
  *     ref or worry about races.
  */
-static void
+static int
 __delete_redo_window_cleanup_internal(WT_SESSION_IMPL *session, WT_REF *ref)
 {
     WT_REF *child;
+    uint64_t sleep_usecs, yield_count;
+    bool busy;
 
     WT_ASSERT(session, F_ISSET(ref, WT_REF_FLAG_INTERNAL));
     if (ref->page != NULL) {
         WT_INTL_FOREACH_BEGIN (session, ref->page, child) {
-            if (child->state == WT_REF_DELETED && child->page_del != NULL)
+            /*
+             * We used to do the cleanup when the state of the child was WT_REF_DELETED, but it can
+             * also be loaded in memory if the page has been read before, thus we always clean the
+             * page_del structure if it has data.
+             */
+            if (child->page_del != NULL) {
                 __cell_redo_page_del_cleanup(session, ref->page->dsk, child->page_del);
+                /* Move back the page to WT_REF_DELETED by marking it for eviction. */
```

Comment:
This is trying to evict not just marking it for eviction

Reply:
Can we add more comments to explain how this fixes the problem?

Reply:
Sure, see 255f0e2ca2e1709427f0c49b430c3f45f1da6917

Reply:
Can we add an assert to ensure the page is clean and we are only doing a clean eviction.

Reply:
Added a check here 5ed9838e3b5650698bfc60b4b800110911676205

Reply:
Given this [comment](https://github.com/wiredtiger/wiredtiger/pull/8416#discussion_r1008973102), the page is not clean, right?

Reply:
I discussed with @kommiharibabu:
```
Before we come to this function, we stop all the eviction threads and perform a checkpoint. This will mark the page as clean.
```

Reply:
Updated in 0fe3bf5660c81f35fe8a31209571e7e70034b944